core/src/apps/cardano/sign_tx/signer.py

```diff
@@ -0,0 +1,903 @@
+            )
+
+        if self.msg.minting_asset_groups_count > 0:
+            minting_dict: HashBuilderDict[bytes, HashBuilderDict] = HashBuilderDict(
+                self.msg.minting_asset_groups_count,
+                wire.ProcessError("Invalid mint token bundle"),
+            )
+            with self.tx_dict.add(TX_BODY_KEY_MINT, minting_dict):
+                await self._process_minting(minting_dict)
+
+        if self.msg.script_data_hash is not None:
+            await self._process_script_data_hash()
+
+        if self.msg.collateral_inputs_count > 0:
+            collateral_inputs_list: HashBuilderList[
+                tuple[bytes, int]
+            ] = HashBuilderList(self.msg.collateral_inputs_count)
+            with self.tx_dict.add(
+                TX_BODY_KEY_COLLATERAL_INPUTS, collateral_inputs_list
+            ):
+                await self._process_collateral_inputs(collateral_inputs_list)
+
+        if self.msg.required_signers_count > 0:
+            required_signers_list: HashBuilderList[bytes] = HashBuilderList(
+                self.msg.required_signers_count
+            )
+            with self.tx_dict.add(TX_BODY_KEY_REQUIRED_SIGNERS, required_signers_list):
+                await self._process_required_signers(required_signers_list)
+
+        if self.msg.include_network_id:
+            self.tx_dict.add(TX_BODY_KEY_NETWORK_ID, self.msg.network_id)
+
+    def _validate_tx_init(self) -> None:
+        if self.msg.fee > LOVELACE_MAX_SUPPLY:
+            raise wire.ProcessError("Fee is out of range!")
+        validate_network_info(self.msg.network_id, self.msg.protocol_magic)
+
+    async def _show_tx_init(self) -> None:
+        if not self._is_network_id_verifiable():
+            await layout.warn_tx_network_unverifiable(self.ctx)
+
+    async def _confirm_tx(self, tx_hash: bytes) -> None:
+        is_network_id_verifiable = self._is_network_id_verifiable()
+        await layout.confirm_tx(
+            self.ctx,
+            self.msg.fee,
+            self.msg.network_id,
+            self.msg.protocol_magic,
+            self.msg.ttl,
+            self.msg.validity_interval_start,
+            is_network_id_verifiable,
+            tx_hash=None,
+        )
+
+    # inputs
+
+    async def _process_inputs(
+        self, inputs_list: HashBuilderList[tuple[bytes, int]]
+    ) -> None:
+        for _ in range(self.msg.inputs_count):
+            input: messages.CardanoTxInput = await self.ctx.call(
+                messages.CardanoTxItemAck(), messages.CardanoTxInput
+            )
+            self._validate_input(input)
+            await self._show_input(input)
+            inputs_list.append((input.prev_hash, input.prev_index))
+
+    def _validate_input(self, input: messages.CardanoTxInput) -> None:
+        if len(input.prev_hash) != INPUT_PREV_HASH_SIZE:
+            raise wire.ProcessError("Invalid input")
+
+    async def _show_input(self, input: messages.CardanoTxInput) -> None:
```

Comment:
Yes, that would require less overrides. The original idea behind this was that the `Signer` would allow everything (so that the child classes can add their own, more strict validation), but also show everything that can be shown (and the child classes can hide some stuff) - it seems natural to me this way ("by default, everything is allowed and shown, and the signing modes can hide something at the expense of forbidding some items / item combinations"). But yes, maybe we can prefer being pragmatic here?